src/main/java/com/minecolonies/coremod/proxy/ClientProxy.java

```diff
@@ -247,6 +246,61 @@ public static void registerModels(@NotNull final ModelRegistryEvent event)
             ModelLoader.setCustomModelResourceLocation(Item.getItemFromBlock(ModBlocks.blockPaperWall), type.getMetadata(),
               new ModelResourceLocation(ModBlocks.blockPaperWall.getRegistryName() + "_" + type.getName(), INVENTORY));
         }
+
+        ModelLoader.setCustomModelResourceLocation(Item.getItemFromBlock(ModBlocks.blockTimberFrameOak), 0,
+                new ModelResourceLocation(new ResourceLocation(Constants.MOD_ID,
+                        BlockTimberFrame.BLOCK_NAME + "_" + BlockPlanks.EnumType.OAK.getName()), INVENTORY));
+
+        ModelLoader.setCustomModelResourceLocation(Item.getItemFromBlock(ModBlocks.blockTimberFrameBirch), 0,
```

Comment:
I think we can store all the ModBlocks.blockTimberFrame in a list in the ModBlocks:

List<BlockTimberFrame> timberFrames = new ArrayList<>();

then we add them were we register them. timberFrames.add

and then we use here a for(final BlockTimberFrame frame: ModBlocks.timberFrames)

to go through all of them and to reduce all of this to a nice line

Reply:
I was actually thinking. With some technology from SmithsCore (just copy) we can make one model, one block and one texture for all wood types registered.

Would that not be much better?

Reply:
that'd be awesome, are you talking about the same tech you're using for the armor?

Reply:
Yes.
The principle is very simple: Before texture stitch you collect all the texture in grayscale you need (so for us basically the parts that are wood), then when the textures get stitched, you ask for the plank texture from the stitcher, replace every grayscale texture with the correct one. And stitch it the new texture in.

Then during model baking you make sure to bake a model for each of the wood types with a correct texture. And when the model is  requested you return the "submodel" with the correct wood type.

The complicated part is the texture stitching.
Before this was an event (very handy) but they moved it so it does not really work anymore. They introduced a new way to make this work using the json file. But that is seriously stupid and very complicated especially if you want to combine it properly with an api. (even made a PR for it......). Right now SmithsCore uses a simple Mixin (thanks @Asherslab) for that btw, works brilliantly simple) that injects my core code into mc that basically fires a custom event in the old spot and makes the behaviour possible again.

Reply:
Some brain food:
- The mixin: https://github.com/Minecolonies/SmithsCore/tree/Alpha/1.11/src/main/com/smithsmodding/smithscore/core
- The texture stitch mechanism:
https://github.com/Minecolonies/SmithsCore/tree/Alpha/1.11/src/main/com/smithsmodding/smithscore/client/textures

Reply:
Then a model loader would be required to redirect MC to our custom model, instead of the default block model. And both an unbaked and a baked custom model which can handle the different wood types, as well as a none ticking TE to store the wood type.

Reply:
Would shrink it down to one block as well
As you could use metadata to store the type.

Reply:
I'd be more than happy to add the mixin part for you if need be mine shadow, it can be a little.... interesting getting the dependency working (for mixins itself), but the classes should be 98% the same as orion's XD

Reply:
I talked to ray about this. Since @MineShadow7 is not experienced enough with the model System we leave it like this.